konfidency/loaders/productDetailsPage.ts

```diff
@@ -0,0 +1,54 @@
+import { AppContext } from "../mod.ts";
+import { ProductDetailsPage } from "../../commerce/types.ts";
+import { ExtensionOf } from "../../website/loaders/extension.ts";
+import { toReview } from "../utils/transform.ts";
+
+export interface Props {
+  /**
+   * @description The default value is 5
+   */
+  pageSize?: number;
+  /**
+   * @description The default value is 1
+   */
+  page?: number;
+}
+
+export default function productDetailsPage(
+  { pageSize = 5, page = 1 }: Props,
+  _req: Request,
+  ctx: AppContext,
+): ExtensionOf<ProductDetailsPage | null> {
+  const { api, customer } = ctx;
+  return async (productDetailsPage: ProductDetailsPage | null) => {
+    if (!productDetailsPage) {
+      return null;
+    }
+
+    if (!productDetailsPage.product.inProductGroupWithID) {
+      return { ...productDetailsPage };
+    }
+
+    try {
+      const reviews = await api["GET /:customer/:sku/summary"]({
+        customer,
+        page,
+        pageSize,
+        sku: productDetailsPage.product.inProductGroupWithID,
+      }).then((res) => res.json());
+      const { aggregateRating, review } = toReview(reviews.reviews[0]);
+
+      return {
+        ...productDetailsPage,
+        product: {
+          ...productDetailsPage.product,
+          aggregateRating,
+          review,
+        },
+      };
+    } catch (error) {
+      console.log(error);
+      return { ...productDetailsPage };
```

Comment:
why did you need to shallow copy here?

Reply:
To return the productDetails without review. I think is better render the PDP component with or without review.